_posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown

```diff
@@ -0,0 +1,77 @@
+
+ABC templates make the indexes use-case-aware and apply the best configuration applicable in terms of performance and end-to-end support at all stages, from index creation to data visualization (for example, Simple Schema for Observability [SS4O]). You can rely on composable index templates and component templates for providing the basic building blocks for the applicable configurations. ABC templates use component templates as a resource in order to expose use-case-specific configurations. The solution is designed to work well with existing index definitions, while the available optimizations are provided straight out of the box.
+
+
+## Why use ABC templates?
+
+Using ABC templates in OpenSearch offers the following advantages:
+
+- **Simplified configuration**: ABC templates eliminate the need to navigate through numerous settings, reducing the complexity of index configuration.
+- **Optimized performance**: The predefined settings and mappings in the templates are optimized for specific use cases, ensuring better performance out of the box.
+- **Automatic updates**: As OpenSearch introduces new optimizations and features, they are added to the ABC template, and you can start using them for their newly created indexes with minimal effort.
+
+
+To understand the optimized settings by ABC templates and their performance benefits, we performed a comparison exercise using an index which relies on the `logs` template using HTTP logs dataset (at the same refresh interval), and found the following:
+
+- Storage improved by 20% (when using `zstd_no_dict` compared to the default `qat_lz4` compression for stored fields).
+- Indexing p99 latency improved by 6% because the generated segments are smaller and thus require fewer merges.
+- The template enforces that the index uses the `log_byte_size` merge policy, thus ensuring that the data ingested together is kept together even after segment merges, resulting in an improved performance of time range filter queries.
+
+The following diagram compares storage and p99 latency performance.
+
+![Application-based template performance comparison](/assets/media/blog-images/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates/perf-comparison.png)
+
+## How to use ABC templates?
+
+To get started with ABC templates, see [Index context](https://opensearch.org/docs/latest/im-plugin/index-context/).
+
+The following animation illustrates creating an index using ABC templates. The template sets the required mappings and settings to the index. Additionally, it configures direct integration with OpenSearch dashboards for the data ingested thereafter.
+
```

Comment:
Line 51: Confirm that my change to "an ABC template" is correct (following "The template" in the next sentence).